agent/api/task.go

```diff
@@ -674,3 +678,27 @@ func (t *Task) String() string {
 	}
 	return res + "]"
 }
+
+// GetID is used to retrieve the taskID from taskARN
+// Reference: http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arn-syntax-ecs
+func (task *Task) GetID() (string, error) {
```

Comment:
Neat function! The error messages could offer more hints about what the underlying problem is. Maybe explicitly say that the arn associated with the task is malformed.

ex: `resource missing delimiter` isn't going to be clear in a logfile without the context of the surrounding functions.